cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -9,10 +9,21 @@ KUBECONFIG_PATH="${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig"
 
 MASTER_NODE="${CLUSTER_NAME}-control-plane"
 WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
+NUM_PFS_REQUIRED="${NUM_PFS_REQUIRED:-1}"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 SRIOV_OPERATOR_NAMESPACE="sriov-network-operator"
 
+decimal_re='^[0-9]+$'
+if ! [[ "$NUM_PFS_REQUIRED" =~ ${decimal_re} ]]; then
+  echo "FATAL: Wrong value of NUM_PFS_REQUIRED, must be numeric, less or equal to actual PF available"
+  exit 1
+fi
+
+if [ "$NUM_PFS_REQUIRED" -eq "0" ]; then
+  echo "NUM_PFS_REQUIRED equals zero, allocating all available PFs to the node"
+fi
```

Comment:
This is wired behavior it could mislead what one would expect to happen
In case `NUM_PFS_REQUIRED` is zero the script should exit, please change it.
Also drop the surrounding quotes there is no need for them when you comparing integers.

Reply:
This is in case we want to give all PFs available
without the need to check how many PFs actually available (some nodes might have 2 some might have 4).

bash should use quotes even if we compare integers, this is the right convention, else there are globbing and word splitting.
(for the var, and for the 0 itself which both ways work for the scalar)

Reply:
The assumption that 0 means all PFs is misleading, this need to be changed.

At this script abstraction level we shouldn't worry about how many PF's there are on a node, 
it should move only what is requested.

About the quotes, why there will be word splinting its a flag that you expect to have an integer
if it doesn't it will fail before.

Reply:
Its not an assumption, its a feature for debugging.
See please the code, and what happens if you put 0 there.
In case of a problem we might do care how many we want to move, and to be agnostic about which baremetal we run on, and how many PFs there are.

About the quotes, its a convention, if you use bash vars you should use quotes, not just if there can be problems.

Reply:
If I wouldn't review the code I wouldn't say anything :)

My point is by doing that you forcing one flag to do two things which is wrong in my opinion.
At the current state we move all PF's but use only one,
with this change we would move one PF which I understand.
But how is moving all PF's to a single namespace will help debugging (or anything)?

About the convention its not craved in stone at least on kubevirtci, 
if you sure a variable will be single "word" there is no need to add quotes.
Anyway this is not the place to discuss what convention to use.

Reply:
> But how is moving all PF's to a single namespace will help debugging (or anything)?

Lets say we there is a bug on CI, that only on one node there is a failure,
and one want to check if its related that we move just one PF to that node,
so just setting it to 0 will allow this to auto detect how many PFs there are and move them, in order to see the behavior.
I saw other problems (not due to this PR), that a feature like this would help to debug if they happened on CI.

There is an echo that explains it, and sometimes a special value like 0 means special things,
for example in timeouts it sometimes infinity, on a non optional fields.

Lets hear another opinion please, and take the majority.
Maybe they will agree with you, but i want to give it a chance.
Same for the quotes.